### PR TITLE
[CARBONDATA-1086] updated configuration-parameters.md and dml-operation-on-carbondata for SORT_SCOPE

### PR DESCRIPTION
**Pull request by vandana7**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3236/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/641/

**Comment by PallaviSingh1992** [comment]
sort.inmemory.size.inmb the default value is 1024 as per the code

**Comment by vandana7** [comment]
@sgururajshetty i have updated the required changes please review

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/695/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3290/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/706/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3302/

**Comment by sgururajshetty** [comment]
LGTM
@chenliang613 kindly review and merge.

File: docs/dml-operation-on-carbondata.md [hunk 1]
```diff
@@ -149,6 +149,50 @@ You can use the following options to load data:
...
+
+    ```
+    OPTIONS('SORT_SCOPE'='BATCH_SORT', 'batch_sort_size_inmb'='7')
+    ```
+
+    Note :
+
+    * batch_sort_size_inmb : Size of data in MB to be processed in batch. By default it is the 45 percent size of sort.inmemory.size.inmb(Memory size in MB available for in-memory sort).
+
+    For GLOBAL_SORT :
```
Comment (zzcclp) [inline docs/dml-operation-on-carbondata.md]: Suggestion: add below note:
`'SINGLE_PASS' must be false.`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3307/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/711/

File: docs/dml-operation-on-carbondata.md (hunk 1 above)
Comment (vandana7) [inline docs/dml-operation-on-carbondata.md]: Hi,
I tried to execute the LOAD query with single_pass= 'true' and sort_scope='BATCH_SORT', it successfully executed  and i was able to fetch the records in sorted way
syntax i used to execute load query - LOAD DATA INPATH 'hdfs://localhost:54310/uniqdata/2000_UniqData.csv' into table uniqdata OPTIONS('DELIMITER'=',' , 'QUOTECHAR'='"','BAD_RECORDS_ACTION'='FORCE','FILEHEADER'='CUST_ID,CUST_NAME,ACTIVE_EMUI_VERSION,DOB,DOJ,BIGINT_COLUMN1,BIGINT_COLUMN2,DECIMAL_COLUMN1,DECIMAL_COLUMN2,Double_COLUMN1,Double_COLUMN2,INTEGER_COLUMN1','SINGLE_PASS'='TRUE','SORT_SCOPE'='BATCH_SORT','batch_sort_size_inmb'='7');


Please let me know if i am doing anything wrong

Comment (zzcclp) [inline docs/dml-operation-on-carbondata.md]: I mean that if SORT_SCOPE=GLOBAL_SORT,single_pass must be false

Comment (zzcclp) [inline docs/dml-operation-on-carbondata.md]: Sorry for my mistake, now when sort_scope='GLOBAL_SORT', single_pass can be 'true',  I have raised a pr to remove this restriction of code (PR-1224).

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

File: docs/dml-operation-on-carbondata.md [hunk 2]
```diff
@@ -149,6 +149,50 @@ You can use the following options to load data:
...
    
+- **SORT_SCOPE:** This property can have four possible values :
+
+    * BATCH_SORT : The sorting scope is smaller and more index tree will be created,thus loading is faster but query maybe slower.
+
+    * LOCAL_SORT : The sorting scope is bigger and one index tree per data node will be created, thus loading is slower but query is faster.
+
+    * GLOBAL_SORT : The sorting scope is bigger and one index tree per task will be created, thus loading is slower but query is faster.
+
+    * NO_SORT     : Feasible if we want to load our data in unsorted manner.
```
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: Introduce this first

File: docs/dml-operation-on-carbondata.md [hunk 3]
```diff
@@ -149,6 +149,50 @@ You can use the following options to load data:
    
    * If this option is set to TRUE, then high.cardinality.identify.enable property will be disabled during data load.
    
+- **SORT_SCOPE:** This property can have four possible values :
+
+    * BATCH_SORT : The sorting scope is smaller and more index tree will be created,thus loading is faster but query maybe slower.
```
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: mention this is sort based on memory size, and carbon will create one index for each batch

File: docs/dml-operation-on-carbondata.md [hunk 4]
```diff
@@ -149,6 +149,50 @@ You can use the following options to load data:
...
+
+    * LOCAL_SORT : The sorting scope is bigger and one index tree per data node will be created, thus loading is slower but query is faster.
+
+    * GLOBAL_SORT : The sorting scope is bigger and one index tree per task will be created, thus loading is slower but query is faster.
```
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: not per task, it is sorting in whole cluster

File: docs/dml-operation-on-carbondata.md (hunk 2 above)
Comment (jackylk) [inline docs/dml-operation-on-carbondata.md]: not we

**Comment by ravipesala** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3371/

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/774/

**Comment by ravipesala** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3372/

**Comment by ravipesala** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/775/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/33/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1212/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/581/

**Pull request closed without merging**